lib/chef-vault/item_keys.rb

```diff
@@ -79,9 +128,51 @@ def save(item_id = @raw_data["id"])
             chef_data_bag.create
           end
         end
+      end
 
+      # write cached keys to data
+      @cache.each do |key, val|
+        spath = "#{@raw_data["id"]}_key_#{key}"
+        # delete across all modes on key deletion
+        if val == false
+          # sparse mode key deletion
+          if Chef::Config[:solo_legacy_mode]
+            FileUtils.rm("#{spath}.json") if File.exist?("#{spath}.json")
+          else
+            begin
+              Chef::DataBagItem.destroy(@data_bag, spath)
+            rescue Net::HTTPServerException => http_error
+              nil if http_error.response.code == "404"
```

Comment:
I don't understand why you specify nil here since this can't be a returned value